src/rust/ide/src/double_representation/refactorings/collapse.rs

```diff
@@ -364,27 +367,35 @@ mod tests {
 
     impl Case {
         fn run(&self, parser:&Parser) {
-            let ast   = parser.parse_module(self.initial_method_code,default()).unwrap();
-            let main  = module::locate_child(&ast,&self.refactored_name).unwrap();
-            let graph = graph::GraphInfo::from_definition(main.item.clone());
-            let nodes = graph.nodes();
-
-            let selected_nodes = nodes[self.extracted_lines.clone()].iter().map(NodeInfo::id);
-            let new_name       = self.introduced_name.clone();
-            let collapsed      = collapse(&graph,selected_nodes,new_name,parser).unwrap();
-
-            let new_method = collapsed.new_method.ast(0,parser).unwrap();
-            let placement  = module::Placement::Before(self.refactored_name.clone());
-            let new_main   = &collapsed.updated_definition.ast;
-            println!("Generated method:\n{}",new_method);
-            println!("Updated method:\n{}",new_main);
-            let mut module = module::Info{ast};
-            module.ast     = module.ast.set(&main.crumb().into(),new_main.ast().clone()).unwrap();
-            module.add_method(collapsed.new_method,placement,parser).unwrap();
-            println!("Module after refactoring:\n{}",&module.ast);
-
-            assert_eq!(new_method.repr(),self.expected_generated);
-            assert_eq!(new_main.repr(),self.expected_refactored);
+            let ast          = parser.parse_module(self.initial_method_code,default()).unwrap();
+            let main         = module::locate_child(&ast,&self.refactored_name).unwrap();
+            let graph        = graph::GraphInfo::from_definition(main.item.clone());
+            let nodes        = graph.nodes();
+            let run_internal = |selection:&Vec<node::Id>| {
+                let selection  = selection.iter().copied();
+                let new_name   = self.introduced_name.clone();
+                let collapsed  = collapse(&graph,selection,new_name,parser).unwrap();
+                let new_method = collapsed.new_method.ast(0, parser).unwrap();
+                let placement  = module::Placement::Before(self.refactored_name.clone());
+                let new_main   = &collapsed.updated_definition.ast;
+                println!("Generated method:\n{}", new_method);
```

Comment:
Done.
It doesn't really make a difference here, as this is test code and these logs are looked upon only when someone is intentionally debugging this. 
But for consistency it might be better to use the logger.